### PR TITLE
Switch s3 bucket to module

### PR DESCRIPTION
**Pull request by kevcube**
## what
* remove unused template provider for Mac m1 compatibility (Implemented by PR https://github.com/cloudposse/terraform-aws-ecs-codepipeline/pull/94) 
* switch s3 bucket `resource` to `module` to fix tests (aws provider v4)

### CODE REVIEW

**Review by nitrocode — DISMISSED** [review]

**Comment by kevcube** [comment]
@nitrocode I have updated the PR title, description and purpose

**Review by nitrocode — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -6,9 +6,12 @@ module "codepipeline_label" {
   context = module.this.context
 }
 
-resource "aws_s3_bucket" "default" {
+module "artifact_store_bucket" {
```
Comment (nitrocode) [inline main.tf:9]: Hmmm this is a bit of scope creep. I don't think we want to use the s3 bucket module in this PR. Let's keep that separated.

**Review by nitrocode — DISMISSED** [review]
Please see comment
